### ISSUE TITLE
Cordova and jQuery still reporting invalid certificate

### ISSUE THREAD

**Issue by Fmstrat** [hypery2k/cordova-certificate-plugin#1]
Hi,

Using the following url, with an invalid certificate: https://192.168.3.201/owncloud

I'm setting the insecure variable to "true", but I'm still getting invalid certificate errors. Here is my code:

```
cordova.plugins.certificates.trustUnsecureCerts(true);
$.ajax({
        type: "POST",
        url: url,
        dataType: 'json',
        async: false,
        cache: false,
        data: d,
        headers: {
                "Authorization": "Basic " + btoa(username + ":" + password)
        },
        success: function(data) {
                eval(f+"(data)");
        },
        error: apiError
});
```

Any thoughts on what could be wrong? I installed the plugin just as described in the README, and everything seemed to work fine.

**Comment by hypery2k** [comment]
in which environment are you testing? within local browser or within a remote debugging session on iOS or Android?

**Comment by Fmstrat** [comment]
OK, I JUST narrowed this down, but now I'm getting the inverse error. So the problem was that ownCloud was not set to use the IP as a trusted domain, so it was throwing a 400 error, which just happens to be the same error you get with an invalid SSL cert. I'm having the exact opposite issue now, everything is trusted. So, if I set the above to "false" it still allows the call to succeed.

I'm testing in Android, using "cordova emulate android" and the "debuggable=true" is NOT set. Are there any required steps like reloading the html page once the true/false flag is set?

_edit_ Interesting, despite debuggable not being true, in release mode in now throws an error 0.

**Comment by hypery2k** [comment]
no, as long as you first set it and afterwards use the AJAX call. But I'm not sure if the Android emulator works correctly with this setup. I've tested it an Android 4.4.2/Android 5.0 and iOS 6 - iOS 8 devices

**Comment by Fmstrat** [comment]
OK, got a step farther, logcat wasn't showing this on the emulator, but on a device:

```
D/PluginManager( 9669): exec() call to unknown plugin: CDVCertificate
```

It seems that despite installing through the "cordova plugin install" it's not finding the plugin.

**Comment by hypery2k** [comment]
ok, I will investigate on this issue. Keeping you updated

**Comment by Fmstrat** [comment]
So far I've discovered it's because of the inconsistencies in plugin naming between ios and android. I created a hack fix in www/certificate.js by changing 'CDVCertificate' to 'CertificatesPlugin' (which would break iOS, but make Android work. This allowed me to get farther, but there is still an error:

```
D/Certificates(13953): Setting allowUntrusted to true
W/WebView (13953): java.lang.Throwable: A WebView method was called on thread 'JavaBridge'. All WebView methods must be called on the same thread. (Expected Looper Looper (main, tid 1) {19154a80} called on Looper (JavaBridge, tid 41109) {20ef190d}, FYI main Looper is Looper (main, tid 1) {19154a80})
W/WebView (13953):      at android.webkit.WebView.checkThread(WebView.java:2194)
W/WebView (13953):      at android.webkit.WebView.setWebViewClient(WebView.java:1672)
W/WebView (13953):      at org.apache.cordova.CordovaWebView.setWebViewClient(CordovaWebView.java:327)
W/WebView (13953):      at de.martinreinhardt.cordova.plugins.CertificatesPlugin.execute(CertificatesPlugin.java:83)
W/WebView (13953):      at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:84)
W/WebView (13953):      at org.apache.cordova.PluginManager.exec(PluginManager.java:151)
W/WebView (13953):      at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:63)
W/WebView (13953):      at org.apache.cordova.ExposedJsApi.exec(ExposedJsApi.java:40)
W/WebView (13953):      at com.android.org.chromium.base.SystemMessageHandler.nativeDoRunLoopOnce(Native Method)
W/WebView (13953):      at com.android.org.chromium.base.SystemMessageHandler.handleMessage(SystemMessageHandler.java:28)
W/WebView (13953):      at android.os.Handler.dispatchMessage(Handler.java:102)
W/WebView (13953):      at android.os.Looper.loop(Looper.java:135)
W/WebView (13953):      at android.os.HandlerThread.run(HandlerThread.java:61)
E/PluginManager(13953): Uncaught exception from plugin
```

As a note, I tried a more formal fix of changing the plugin reference in plugin.xml, but it wouldn't stick when compiling with "cordova build android --release". I have no knowledge of plugin development, though, so it's probably just something I'm doing wrong.

_edit_ Added the full error since it shows the line numbers.

**Comment by Fmstrat** [comment]
Fixed both issues, and created a pull request that I think solves all the issues. Look it over though, as I've never built a plugin and am fairly thread-stupid. https://github.com/hypery2k/cordova-certificate-plugin/pull/2/files

**Comment by hypery2k** [comment]
fixed with version 0.4.0, see http://plugins.cordova.io/#/package/de.martinreinhardt.cordova.plugins.certificates
